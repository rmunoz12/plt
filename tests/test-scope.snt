@setup {
    int x = 4;

    group A() {
        int x;

        func group A __init__(int x) {
            this.x = x;
            return this;
        }

        func void x() {

        }
    };

    func str test() { return "test0"; }
    func str test(int x) { return "test1"; }

}

@turns {

func void begin() {
    int x = 6;
    group A a = A(25);

    print(x); print("\n");
    print(a.x); print("\n");
    print(test()); print("\n");
    print(test(5)); print("\n");
    end;
}

}
