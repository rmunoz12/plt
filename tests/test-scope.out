6
25
test0
test1
